The --without-opensc logic is broken, so let's fix it.

patch by Stian Skjelstad.

http://bugs.gentoo.org/show_bug.cgi?id=78730

--- openssh-3.9p1/configure.ac
+++ openssh-3.9p1/configure.ac
@@ -2171,9 +2171,9 @@
 AC_ARG_WITH(opensc,
 	AC_HELP_STRING([--with-opensc=PFX],
 		       [Enable smartcard support using OpenSC]),
-	opensc_config_prefix="$withval", opensc_config_prefix="")
-if test x$opensc_config_prefix != x ; then
-  OPENSC_CONFIG=$opensc_config_prefix/bin/opensc-config
+		       [
+if test "x$withval" != xno ; then
+  OPENSC_CONFIG="$withval/bin/opensc-config"
   AC_PATH_PROG(OPENSC_CONFIG, opensc-config, no)
   if test "$OPENSC_CONFIG" != "no"; then
     LIBOPENSC_CFLAGS=`$OPENSC_CONFIG --cflags`
@@ -2183,8 +2183,12 @@
     AC_DEFINE(SMARTCARD)
     AC_DEFINE(USE_OPENSC)
     SCARD_MSG="yes, using OpenSC"
+  else
+   AC_MSG_ERROR([opensc-config not found])
   fi
 fi
+			]
+)
 
 # Check libraries needed by DNS fingerprint support
 AC_SEARCH_LIBS(getrrsetbyname, resolv,
